force-app/main/default/classes/DMLOperations.cls

```diff
@@ -144,33 +171,54 @@ public with sharing class DMLOperations {
      * @param accountName The name of the Account.
      * @param oppNames The list of Opportunity names.
      */
-    public static void upsertOpportunities(String accountName, List<String> oppNames) {
-        // Loop through each Opportunity name
-            // Create an Opportunity with the given name
-            // Add the Opportunity to the list
-    }
+        public static void upsertOpportunities(String accountName, List<String> oppNames) {
+            // Loop through each Opportunity name
+            List<Opportunity> opportunityName = new List <Opportunity>();
+            Account account = new Account (Name = accountName);
+            insert account;
+            for(String name : oppNames){
+                // Create an Opportunity with the given name
+                Opportunity myOpp = new Opportunity();
+                myOpp.Name = name;
+                myOpp.Amount = 50000;
+                myOpp.StageName = 'Qualification';
+                myOpp.CloseDate = Date.today().addMonths(3);
+                myOpp.AccountId = account.Id;
+                // Add the Opportunity to the list
+                opportunityName.add(myOpp);
+            }
+        upsert opportunityName;
+        }
```

Comment:
This method isn't passing the test at the moment. Go back and read the instructions. As a hint, you'll need to (1) determine if the Account already exists, and (2) determine if any of the Opportunities already exist. Run the test method to check which errors it is throwing. If you're not sure what I'm talking about, respond here or reach out on Slack.